### DISCUSSION TITLE
Checks fail when adding new language (Open SDG Montenegro)

### DISCUSSION THREAD

**Discussion by nikolabul** [open-sdg/open-sdg#1730] [Q&A] [answered]
Hi everyone. I am developing the Open SDG platform in Montenegro and I have a problem with adding the Montenegrin language to the platform. Montenegrin language is not present in the Webplate platform (I made a request) so I tried the steps below.

Steps in data/:
- added Montenegrin ("me") language in data/config_data.yml
- added "me" folder in data/translations and some custom .yml files

Steps in site/:
- added "me" in site/_data/site_config.yml
- added "me" in site/_config.yml
- added "me" folders and files in _pages and _posts using add_language.py
- added "me" folders and files in assets/img/goals and assets/img/high_contrast/goals
- performed all folder and file changes and additions in one pull-request

Data steps pass all the checks but site checks fail. 

I get this error in the site HTML test:
```
    Ran on 2052 files!
    - ./_test/site/me/search/index.html
      *  image /site/assets/img/loading.gif does not have an alt attribute (line 313)
    
    HTML-Proofer found 1 failure!
    Error: Process completed with exit code 1.
```

Could you suggest where I am making mistake and why my site checks fail?

Thanks

**Comment by brockfanning** [comment, marked as answer]
@nikolabul Thanks for taking the initiative on adding the new language! We have merged the new language and the initial translations into the "2.1.0-dev" branch. So, to use these translations you'll need to make sure your data configuration is pointing to this branch. Here is the line in the "data starter" project that you would need to update: https://github.com/open-sdg/open-sdg-data-starter/blob/develop/config_data.yml#L76 (you would need it to say "2.1.0-dev" instead of "2.0.0" in that example)

The steps that you listed seem correct, but one important note: in the translation platform that we use (Weblate) the language code is "cnr". So you will need to use "cnr" instead of "me" throughout all of those steps.

Let me know if these changes help. There are a lot of moving parts with translations so I may be forgetting something.

**Reply by nikolabul** [reply]
Thanks for suggestions @brockfanning.

I completed the Webplate translations for 'cnr' and changed the data configuration for languages to 2.1.0-dev.
After that, I get the following error during site checks:
```
- ./_test/site/test-post-1/index.html
  *  mailto:%email_contacts.suggestions contains an invalid email address (line 169)
     <a href="mailto:%email_contacts.suggestions">feedback</a>
```

I added email_contacts.suggestions in my site_config.yml and set it to montenegrosdg@gmail.com but I get the same error again.

What could be the problem?

**Comment by nikolabul** [comment]
```
Ran on 20[52](https://github.com/montestat/site/runs/7250737775?check_suite_focus=true#step:7:53) files!
- ./_test/site/1-1-1/config/index.html
  *  mailto:%email_contacts.suggestions contains an invalid email address (line 169)
     <a href="mailto:%email_contacts.suggestions">feedback</a>
- ./_test/site/1-1-1/data/index.html
  *  mailto:%email_contacts.suggestions contains an invalid email address (line 169)
     <a href="mailto:%email_contacts.suggestions">feedback</a>
- ./_test/site/1-1-1/index.html
  *  mailto:%email_contacts.suggestions contains an invalid email address (line 174)
     <a href="mailto:%email_contacts.suggestions">feedback</a>
- ./_test/site/1-1-1/metadata/index.html
  *  mailto:%email_contacts.suggestions contains an invalid email address (line 169)
     <a href="mailto:%email_contacts.suggestions">feedback</a>
- ./_test/site/1-2-1/config/index.html
  *  mailto:%email_contacts.suggestions contains an invalid email address (line 169)
     <a href="mailto:%email_contacts.suggestions">feedback</a>
- ./_test/site/1-2-1/data/index.html
  *  mailto:%email_contacts.suggestions contains an invalid email address (line 169)
     <a href="mailto:%email_contacts.suggestions">feedback</a>
- ./_test/site/1-2-1/index.html
  *  mailto:%email_contacts.suggestions contains an invalid email address (line 174)
     <a href="mailto:%email_contacts.suggestions">feedback</a>
```

**Reply by nikolabul** [reply]
and it goes like this for many pages

**Reply by brockfanning** [reply]
@nikolabul Apologies, when I recommended updating the translations I also should have mentioned that you would need to upgrade to 2.0.0. So you would need to change "1.8.0" to "2.0.0" on these lines:
* https://github.com/montestat/data/blob/develop/scripts/requirements.txt#L2
* https://github.com/montestat/site/blob/develop/_config.yml#L4
* https://github.com/montestat/site/blob/develop/Gemfile#L7

**Reply by nikolabul** [reply]
Thank you @brockfanning. Now the checks pass.

**Comment by nikolabul** [comment]
Hi @brockfanning, I have a new problem with translations. I wanted to make Montenegrin (cnr) language a default one so I made the following changes:

- in languages sections of config_data.yml and site_config.yml I made cnr language the first one:
```
languages:
  - cnr
  - en
```
- I set the structure of _pages/ as follows:
     - /_pages/ consists of Montenegrin versions of 404.md, about.md, categories.md etc.
     - /_pages/en/ consists of English versions of 404.md, about.md, categories.md etc.

- I set the same structure for _posts/ folder


After I run the site tests I get these errors:
```
For the Links check, the following failures were found:
* At ./_test/site/categories/index.html:272:
Ran on 20[52](https://github.com/montestat/site/runs/7414587600?check_suite_focus=true#step:7:53) files!
  'a' tag is missing a reference
* At ./_test/site/en/categories/index.html:272:
  'a' tag is missing a reference
* At ./_test/site/en/index.html:2[56](https://github.com/montestat/site/runs/7414587600?check_suite_focus=true#step:7:57):
 [ http://www.un.org/ga/search/view_doc.asp?symbol=A/RES/70/1&Lang](http://www.un.org/ga/search/view_doc.asp?symbol=A/RES/70/1&Lang=E)=E is not an HTTPS lin
* At ./_test/site/index.html:256:
 [ http://www.un.org/ga/search/view_doc.asp?symbol=A/RES/70/1&Lan](http://www.un.org/ga/search/view_doc.asp?symbol=A/RES/70/1&Lang=)g= E is not an HTTPS lin
```


Do you have any suggestions what am I doing wrong?

**Reply by brockfanning** [reply]
@nikolabul You're not doing anything wrong - that problem is caused because the html-proofer project released a new version that is more strict about certain things. Here is the fix we had to do in the site starter - I would suggest you make the same change in your site repository: https://github.com/open-sdg/open-sdg-site-starter/commit/ed2b6c1460d139c728c559587cfea8cf7f7c25c8

**Reply by nikolabul** [reply]
Great, thanks a lot. Now it works fine.
